### PR TITLE
Add extendable MFD overlay

### PR DESCRIPTION
**Pull request by ashwin-nat**
## 📝 Description

Adding an extendable MFD overlay

---

## 📂 Type of change

- [ ] Bug fix 🐞
- [ ] New feature 🚀
- [ ] Refactor 🔨
- [ ] Documentation 📚
- [ ] Tests ✅
- [ ] Other: 

---

## 🧪 Backend Test Reports (required for Python/backend changes)

> ⚠️ If your PR includes backend Python changes, please fill out the sections below.

### ✅ Integration Tests

* [ ] All integration tests pass
* Attach test command/output:

```
<paste result here>
```

To run the integration test
```bash
poetry run python tests/integration_test/runner.py
```

---

## 🧱 `lib/` Directory Changes

* [ ] This PR modifies or adds files under `lib/`
* [ ] I have updated or added unit tests to reflect those changes

Explain your test updates here:

```
Explain what was added/updated, or write "N/A" if not applicable.
```

---

## 📋 General Checklist

* [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) guide
* [ ] My code follows project style conventions
* [ ] All new and existing tests pass
* [ ] I have added relevant documentation/comments
* [ ] I have reviewed my changes and believe they are ready to merge

---

## Summary by Sourcery

Implement a fully extendable multi-function display (MFD) overlay feature, including new UI pages, configuration flags, UDP‐driven page cycling, backend and IPC integration, and corresponding tests and documentation updates

New Features:
- Add new MFD overlay subsystem with multiple pages (lap times, tyre wear, fuel info, weather forecast, pit rejoin prediction) and animated expand/collapse behavior
- Introduce configuration flags to enable the MFD overlay and assign a UDP action code to cycle its pages
- Implement backend handling of a cycle-MFD UDP action, inter-task notification, and IPC command to trigger page changes
- Stream overlay updates via a new `stream-overlay-update` event in the telemetry pipeline and HUD client listener
- Add F1Utils.getMaxTyreWear utility and integrate maximum tyre wear display in overlays

Enhancements:
- Unify validation to ensure all UDP action codes across config sections are unique

Documentation:
- Revise architecture diagram in docs/arch.mmd to reflect updated launcher, backend, and overlay flows

Tests:
- Add tests for new HUD settings (`show_mfd`, `cycle_mfd_udp_action_code`) and for duplicate UDP action code validation

Chores:
- Rename HUD manager’s ping button to hide/show and add temporary next page button implementation in launcher
- Clean up debug logging in the window manager

### CODE REVIEW

File: apps/hud/ipc/handlers.py [hunk 1]
```diff
@@ -89,3 +89,20 @@ def handle_set_opacity(msg: dict, logger: logging.Logger, overlays_mgr: Overlays
...
+def handle_next_page(msg: dict, logger: logging.Logger, overlays_mgr: OverlaysMgr) -> dict: # TODO - remove
+    """Handle the 'next-page' IPC command to show next page of HUD widgets.
+
+    Args:
+        msg (dict): IPC command message
+        logger (logging.Logger): Logger
+        overlays_mgr (OverlaysMgr): Overlays manager
+
+    Returns:
+        dict: IPC response
```
[the remark this answers is not in the document]
Comment (ashwin-nat) [reply apps/hud/ipc/handlers.py:102]: this is merging into a dev branch

File: apps/hud/ui/overlays/mfd/pages/pit_rejoin/pit_rejoin_page.py [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+        # Step 2: Find rejoin position (where projected gap would place them)
+        rejoin_index = len(table_entries) - 1  # assume they rejoin last
+        for i, row in enumerate(table_entries):
+            if projected_gap < row["delta-info"]["delta-to-leader"]:
+                rejoin_index = i - 1
+                break
+
+        if rejoin_index < 0:
+            rejoin_index = 0  # ahead of everyone
+
```
[the remark this answers is not in the document]
Comment (ashwin-nat) [reply apps/hud/ui/overlays/mfd/pages/pit_rejoin/pit_rejoin_page.py:596]: for loop is more readable for me
